gamemodes/terrortown/gamemode/client/cl_search.lua

```diff
@@ -2,723 +2,393 @@
+		buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButton, self.sizes.padding + 1)
+	elseif not bodysearch.CanConfirmBody() then
+		if data.credits > 0 and client:IsActiveShopper() and not clientRD.preventFindCredits then
+			if data.credits == 1 then
+				buttonConfirm:SetText("search_take_credit")
+				buttonConfirm:SetParams({credit = data.credits})
+			else
+				buttonConfirm:SetText("search_take_credits")
+				buttonConfirm:SetParams({credits = data.credits})
+			end
+			buttonConfirm:SetIcon(materialCredits)
+			buttonConfirm:SetSize(self.sizes.widthButtonTakeCredits, self.sizes.heightButton)
+			buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButtonTakeCredits, self.sizes.padding + 1)
+		else
+			buttonConfirm:SetText("search_confirm_forbidden")
+			buttonConfirm:SetEnabled(false)
+			buttonConfirm:SetSize(self.sizes.widthButton, self.sizes.heightButton)
+			buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButton, self.sizes.padding + 1)
+		end
+	elseif data.credits > 0 and playerCanTakeCredits then
+		if data.credits == 1 then
+			buttonConfirm:SetText("search_confirm_credit")
+			buttonConfirm:SetParams({credit = data.credits})
+		else
+			buttonConfirm:SetText("search_confirm_credits")
+			buttonConfirm:SetParams({credits = data.credits})
+		end
+		buttonConfirm:SetSize(self.sizes.widthButtonCredits, self.sizes.heightButton)
+		buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButtonCredits, self.sizes.padding + 1)
+		buttonConfirm:SetIcon(materialCredits)
 	else
-		search.kills = nil
+		buttonConfirm:SetText("search_confirm")
+		buttonConfirm:SetSize(self.sizes.widthButton, self.sizes.heightButton)
+		buttonConfirm:SetPos(self.sizes.widthMainArea - self.sizes.widthButton, self.sizes.padding + 1)
 	end
+	buttonConfirm.DoClick = function(btn)
+		local creditsOnly = playerCanTakeCredits
+			and GetConVar("ttt2_inspect_confirm_mode"):GetInt() > 0
+			and not (clientRD.isPolicingRole and clientRD.isPublicRole)
 
-	search.lastid = {idx = net.ReadUInt(8)}
+		bodysearch.ClientConfirmsCorpse(data.rag, data.searchUID, data.lrng, creditsOnly)
 
-	-- should we show a menu for this result?
-	search.finder = net.ReadUInt(8)
-	search.show = LocalPlayer():EntIndex() == search.finder
+		-- call these functions locally to give the user an instant feedback
+		-- the same function will be called again after the server processed
+		-- the confirmation and reported back to the clients
 
-	--
-	-- last words
-	--
-	local words = net.ReadString()
-	search.words = (words ~= "") and words or nil
-
-	-- long range
-	search.lrng = net.ReadBit()
-
-	-- searched by detective?
-	search.detective_search = net.ReadBool()
-
-	-- set search.show_sb based on detective_search or self search
-	search.show_sb = search.show or search.detective_search
-
-	---
-	-- @realm shared
-	hook.Run("TTTBodySearchEquipment", search, eq)
+		if playerCanTakeCredits then
```

Comment:
![image](https://github.com/TTT-2/TTT2/assets/13639408/61aa42f9-f1a0-4694-a2d5-f3a1548ea9ea)
this is how it is defined. I'd move it to the player class if I'd move it anywhere at all

Reply:
Since you already introduced the `bodysearch.CanConfirmBody()` function, i think this one would also be fine on the bodysearch table.
This would also make it easy to add e.g. a hook later on etc.

Reply:
done